specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.6-preview.1/backuprestore.json

```diff
@@ -82,6 +82,62 @@
         }
       }
     },
+    "/prebackup": {
+      "post": {
+        "tags": [
+          "PreFullBackup"
+        ],
+        "operationId": "PreFullBackup",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "description": "Pre-backup operation for checking whether the customer can perform a full backup operation.",
+        "parameters": [
+          {
+            "in": "body",
+            "name": "preBackupOperationParameters",
+            "schema": {
+              "$ref": "#/definitions/PreBackupOperationParameters"
```

Comment:
Just cross verifying for my understanding, here you are actually creating a similar class as that of backup input params with the additional change of making each of the field nullable to make the params optional correct?

Reply:
Correct. In case of backup endpoint, `storageResourceUri` is a required parameter. Since there are no required parameters for the pre-backup, we needed a separate definition which can optionally support all parameters and validate the same if provided.